sdk/src/clients/gas/DefaultGasProvider.ts

```diff
@@ -59,9 +59,10 @@ export class DefaultGasProvider implements IEthereumGasProvider<TransactionReque
     }
 
     const feeHistory = await this.fetchFeeHistory();
+    const baseFeePerGas = this.calculateBaseFeePerGas(feeHistory.baseFeePerGas);
     const maxPriorityFeePerGas = this.calculateMaxPriorityFee(feeHistory.reward);
 
-    if (maxPriorityFeePerGas > this.config.maxFeePerGasCap) {
+    if (baseFeePerGas + maxPriorityFeePerGas > this.config.maxFeePerGasCap) {
```

Comment:
Why do we need `baseFeePerGas + maxPriorityFeePerGas` here instead of just `maxPriorityFeePerGas`? `baseFeePerGas` is calcuated as average from the `feeHistory.baseFeePerGas` but on line 133 we just use the latest feeHistory baseFeePerGas times 2 as `const maxFeePerGas = BigInt(baseFeePerGas[baseFeePerGas.length - 1]) * 2n + maxPriorityFeePerGas;`

What's the point of calculating the above `baseFeePerGas` as `feeHistory.baseFeePerGas` average and putting it with check above?

Reply:
Yes that's a mistake thanks Jones :)